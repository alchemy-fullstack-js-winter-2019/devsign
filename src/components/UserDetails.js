import React from 'react';
import PropTypes from 'prop-types';
import { Avatar, Aside, Bio } from '../styles/userDetails';
import { Handle } from '../styles/mainStyle';

function UserDetails({ user }) {
  const { name, handle, profileImage, bio, location } = user;
  return (
    <Aside>
      <Avatar name="profileImage" src={profileImage} />
      <h2>{name} <Handle>{handle}</Handle></h2>
      <Bio>{bio}</Bio>
      <Bio>{location}</Bio>
    </Aside>
  );
}

UserDetails.propTypes = {
  user: PropTypes.object.isRequired
};

export default UserDetails;
