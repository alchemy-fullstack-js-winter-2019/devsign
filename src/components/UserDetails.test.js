import React from 'react';
import UserDetails from './UserDetails';
import { shallow } from 'enzyme';

describe('UserDetails component', () => {
  it('matches a snapshot', () => {
    const  user = {
      name: 'Kaiya',
      handle: '@ladybeard',
      profileImage: 'http://capepremierrealty.com/wp-content/uploads/2018/11/generic-profile-icon-7.jpg.png',
      bio: 'Fullstack MERN Developer / Animal Lover / Hiker',
      location: 'Portland, OR'
    };
    const wrapper = shallow(
      <UserDetails 
        user={user}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
