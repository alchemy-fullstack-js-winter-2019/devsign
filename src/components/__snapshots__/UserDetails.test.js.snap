// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails component matches a snapshot 1`] = `
<styled.aside>
  <Styled(styled.img)
    name="profileImage"
    src="http://capepremierrealty.com/wp-content/uploads/2018/11/generic-profile-icon-7.jpg.png"
  />
  <h2>
    Kaiya
     
    <styled.span>
      @ladybeard
    </styled.span>
  </h2>
  <styled.p>
    Fullstack MERN Developer / Animal Lover / Hiker
  </styled.p>
  <styled.p>
    Portland, OR
  </styled.p>
</styled.aside>
`;
